generate/generate.go

```diff
@@ -0,0 +1,72 @@
+package generate
+
+import (
+	"fmt"
+	"crypto/rsa"
+	"crypto/rand"
+	"crypto/x509"
+	"io/ioutil"
+	"encoding/pem"
+)
+
+func generatePrivateKey(bits int) (*rsa.PrivateKey, error){
+	privateKey, err := rsa.GenerateKey(rand.Reader, bits)
+
+	if err != nil {
+		fmt.Println(err.Error)
+		return nil, err
+	}
+
+	return privateKey, nil
+}
+
+func generatePublicKey(privateKey *rsa.PrivateKey) *rsa.PublicKey {
+	publicKey := &privateKey.PublicKey
+	return publicKey
+}
+
+func savePrivateKey(privateKey *rsa.PrivateKey) error {
+	PrivASN1 := x509.MarshalPKCS1PrivateKey(privateKey)
+
+	privBytes := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: PrivASN1,
+	})
+	ioutil.WriteFile("private.key", privBytes, 0644) // FIXME @antoniaklja shall we pass destination path here?
```

Comment:
shouldn't the access be `600` or even `400`?

Reply:
let's use 600 - owner read write